lib/search.js

```diff
@@ -28,7 +29,24 @@ function processAndRecur (html, opts, savedApps, mappings) {
  * the results.
  */
 function checkFinished (opts, savedApps, nextToken) {
+  // loads fullDetail for all apps that are being returned (starting with last item)
+  function load (num) {
```

Comment:
this looks a bit more complicated than it could be in my opinion. Considering we need to call the app method once per each result, I think it would be best to adding something like this at the bottom of the module:

```js
..then((results) => {
    if (opts.fullDetail) {
      return results.map(() => app({...}))
    }
    return results;
  })
```

Also, it's important that when you call the app method you pass the same opts used for the search, so options like language are respected.

Reply:
this really looks easier, I'm gonna try this

Reply:
oh and you probably need to call Promise.all on the resulting list to turn it into alist of results instead of a list of promises